token-swap/program/src/processor.rs

```diff
@@ -294,6 +306,12 @@ impl Processor {
         fees.validate()?;
         swap_curve.calculator.validate()?;
 
+        // Verify that the deposit authority is a signer to
+        // ensure Pool creator can deposit
+        if !deposity_authority_info.is_signer {
+            return Err(SwapError::DepositAuthorityNotSigner.into());
+        }
+
```

Comment:
In general when you set/change an authority for something, solana programs don't require the new authority to be a signer. For example, the  [spl-token Approve instruction] (https://docs.rs/spl-token/3.2.0/spl_token/instruction/enum.TokenInstruction.html#variant.Approve) doesnt require the approved delegate to be a signer. Doing so allows the authority to be set to a PDA without the authorized program having a specialized instruction to make a CPI in order to sign for that PDA. The drawback is if you mess the authority up the authority will be locked forever lol.